tools/benchmark/src/test/Benchmark.unit.tests.ts

```diff
@@ -125,6 +130,79 @@ describe("`benchmark` function", () => {
 			type: BenchmarkType.OwnCorrectness,
 		});
 	}
+
+	// This pattern is roughly what the non-custom `benchmark` does.
+	// It minimizes per iteration over head by timing the whole batch as a single unit.
+	// This is important for timing operations which are very fast relative to measurement overhead and clock precision.
+	// Since measurement overhead and clock precision are not the same on all systems, this approach necessary to be robustly portable.
+	customBenchmark({
```

Comment:
>customBenchmark [](http://example.com/codeflow?start=1&length=15)

Just to brainstorm, have you considered keeping the existing benchmark name and perhaps using a new type to do the logic do-> while below ?  And the original benchmarkFnAsync would be the "do the thing"

Reply:
I don't think I understand your suggestion, but it inspired me to find a way to make the new custom benchmarks more consistent with the other types, which I have now done.